src/core/helpers/index.ts

```diff
@@ -588,3 +588,88 @@ export function getAccountDates(): { startDate: Date; endDate: Date } {
 
   return { startDate, endDate };
 }
+
+export async function checkBuffer(ctx: MyContext, messageIds: number[], amount: number, userPlan: UserPlan): Promise<String | void> {
+  if(userPlan === UserPlan.MEDIUM_RISK || userPlan === UserPlan.LOW_RISK) {
+
+    const buffer = await HinBuffer.findOne();
+    if (buffer) {
+    if ((buffer.amount) * 2 == buffer.amount_allocated) {
+      const reply = await ctx.reply('<b>No more deposits can be made at this time.</b> 🚫', {
+        parse_mode: 'HTML'
+      });
+      messageIds.push(reply.message_id);
+      return "false";
+    }
+    let availableAmount = (((buffer.amount) * 2) - (buffer.amount_allocated)) * 2;
+
+    if (userPlan === UserPlan.MEDIUM_RISK) {
+      if (amount > availableAmount) {
+        const reply = await ctx.reply(`<b>Amount too large.</b> 🚫\n\n Your deposit should not exceed ${formatNumber(availableAmount)}`, {
+          parse_mode: 'HTML'
+        });
+        messageIds.push(reply.message_id);
+        return "Try again";
+      }
+    }
+    else if (userPlan === UserPlan.LOW_RISK) {
+      if (amount > availableAmount / 2) {
+        const reply = await ctx.reply(`<b>Amount too large.</b> 🚫\n\n Your deposit should not exceed ${formatNumber(availableAmount / 2)}`, {
+          parse_mode: 'HTML'
+        });
+        messageIds.push(reply.message_id);
+        return "Try again";
+      }
+    }
+  }
+  else if(!userPlan) {
+    const reply = await ctx.reply('<b>Invalid Plan</b> 🚫\n\nPlease select a valid plan to proceed.', {
+      parse_mode: 'HTML'
+    });
+    messageIds.push(reply.message_id);
+    ctx.session.route = 'choosePlanForDeposit';
+  }
+  console.log('Buffer check passed');
+  }
+}
+
+export async function updateBufferDeposits(ctx: MyContext, messageIds: number[], amount: number, userPlan: UserPlan): Promise<void> {
+  const buffer = await HinBuffer.findOne();
```

Comment:
@CUTolu2021 If we are calling this here, there are two ways to handle it...
- Either pass it as a parameter into the `checkBuffer` function OR
- Call it in the `checkBuffer` function and return an object that has `buffer`

There is no reason why we should query for the buffer twice

Reply:
i use buffer in router/deposit so it meant that checkbuffer has to query for the buffer independently

Reply:
Noted, but that means we can do sth like this,

```suggestion
  const bufferResponse = await checkBuffer(ctx, messageIds, amount, userPlan);
```

and `checkBuffer` will return an object -> {
  buffer: Buffer,
  response: string
}